.tools/convert-markdown-into-ipynb-and-test.sh

```diff
@@ -29,7 +29,19 @@ for file in */{README,README\.en}.ipynb ; do
     cd $(dirname $file) > /dev/null
 
     echo "begin test $file"
-    jupyter nbconvert --to python $(basename $file) --stdout | python
+    if [[ $(dirname $file) == "08.recommender_system" ]]; then
+       timeout -s SIGKILL 30 bash -c \
+           "jupyter nbconvert --to python $(basename $file) --stdout  | \
+           sed  's/get_ipython()\.magic(.*'\''matplotlib inline'\'')/\#matplotlib inline/g' | \
+           sed '/^# coding: utf-8/a\import matplotlib\nmatplotlib.use('\''Agg'\'')' | python"
+    else
+       timeout -s SIGKILL 30  bash -c "jupyter nbconvert --to python $(basename $file) --stdout | python" 
+    fi
+
+    if [[ $? -ne 0 && $? -ne 124 && $? -ne 137 ]]; then
```

Comment:
>If the command times out, and --preserve-status is not set, then exit
       with status 124.  Otherwise, exit with the status of COMMAND.  If no
       signal is specified, send the TERM signal upon timeout.  The TERM
       signal kills any process that does not block or catch that signal.
       It may be necessary to use the KILL (9) signal, since this signal
       cannot be caught, in which case the exit status is 128+9 rather than
       124.  

确实不用写124